src/lib.rs

```diff
@@ -3053,4 +3108,77 @@ mod tests {
         });
         assert!(a.eq_vec(&[false, true, false, true, false, true, false, true]));
     }
+
+    #[test]
+    fn test_insert_at_zero() {
+        let mut v = BitVec::new();
+
+        v.insert(0, false);
+        v.insert(0, true);
+        v.insert(0, false);
+        v.insert(0, true);
+        v.insert(0, false);
+        v.insert(0, true);
+
+        assert_eq!(v.len(), 6);
+        assert_eq!(v.storage().len(), 1);
+        assert!(v.eq_vec(&[true, false, true, false, true, false]));
+    }
+
+    #[test]
+    fn test_insert_at_end() {
+        let mut v = BitVec::new();
+
+        v.insert(v.len(), true);
+        v.insert(v.len(), false);
+        v.insert(v.len(), true);
+        v.insert(v.len(), false);
+        v.insert(v.len(), true);
+        v.insert(v.len(), false);
+
+        assert_eq!(v.storage().len(), 1);
+        assert_eq!(v.len(), 6);
+        assert!(v.eq_vec(&[true, false, true, false, true, false]));
+    }
+
+    #[test]
+    fn test_insert_at_block_boundaries() {
+        let mut v = BitVec::from_elem(32, false);
+
+        assert_eq!(v.storage().len(), 1);
+
+        v.insert(31, true);
+
+        assert_eq!(v.len(), 33);
+
+        assert!(matches!(v.get(31), Some(true)));
+        assert!(v.eq_vec(&[
+            false, false, false, false, false, false, false, false, false, false, false, false,
+            false, false, false, false, false, false, false, false, false, false, false, false,
+            false, false, false, false, false, false, false, true, false
+        ]));
+
+        assert_eq!(v.storage().len(), 2);
+    }
+
+    #[test]
+    fn test_insert_at_block_boundaries_1() {
```

Comment:
We really could benefit a lot in our project from the `test_case` lib dev dependency for tests.